File: src/main/java/guru/xgm/jass/extapi/psi/JassPsiFileBase.java
package guru.xgm.jass.extapi.psi;

import guru.xgm.jass.lang.JassLanguage;
import guru.xgm.jass.openapi.fileTypes.JassLanguageFileType;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import org.jetbrains.annotations.NotNull;

public class JassPsiFileBase extends PsiFileBase {

    public JassPsiFileBase(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, JassLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return JassLanguageFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "JASS File";
    }
}

File: src/main/kotlin/raft/war/language/jass/psi/JassPsiFileBase.kt
package raft.war.language.jass.psi

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider
import raft.war.language.jass.lang.JassLanguage
import raft.war.language.jass.openapi.fileTypes.JassFileType

class JassPsiFileBase(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, JassLanguage.instance) {
    override fun getFileType(): FileType {
        return JassFileType.instance
    }

    override fun toString(): String {
        return JassLanguage.NAME
    }
}
